src/box/sql/select.c

```diff
@@ -2102,55 +2093,45 @@ sqlColumnsFromExprList(Parse * parse, ExprList * expr_list,
 				int iCol = pColExpr->iColumn;
 				assert(iCol >= 0);
 				space_def = pColExpr->space_def;
-				zName = space_def->fields[iCol].name;
+				name = space_def->fields[iCol].name;
 			} else if (pColExpr->op == TK_ID) {
 				assert(!ExprHasProperty(pColExpr, EP_IntValue));
-				zName = pColExpr->u.zToken;
+				name = pColExpr->u.zToken;
 			}
 		}
-		if (zName == NULL) {
+		if (name == NULL) {
 			uint32_t idx = ++parse->autoname_i;
-			zName = sql_xstrdup(sql_generate_column_name(idx));
+			name = sql_xstrdup(sql_generate_column_name(idx));
 		} else {
-			zName = sql_xstrdup(zName);
+			name = sql_xstrdup(name);
 		}
+		size_t len = strlen(name);
 
 		/* Make sure the column name is unique.  If the name is not unique,
 		 * append an integer to the name so that it becomes unique.
 		 */
-		cnt = 0;
-		while (zName && sqlHashFind(&ht, zName) != 0) {
-			nName = sqlStrlen30(zName);
-			if (nName > 0) {
-				int j;
-				for (j = nName - 1;
-				     j > 0 && sqlIsdigit(zName[j]); j--);
-				if (zName[j] == '_')
-					nName = j;
-			}
-			zName = sqlMPrintf("%.*z_%u", nName, zName, ++cnt);
-		}
-		assert(zName != NULL);
```

Comment:
And here you remove the assertion added in commit `sql: drop check of zName in sqlColumnsFromExprList`... Please squash the two patches.

Reply:
Done.